public/app/features/dashboard/dashgrid/DashboardGrid.tsx

```diff
@@ -108,7 +108,9 @@ export class DashboardGrid extends PureComponent<Props, State> {
   };
 
   onResize: ItemCallback = (layout, oldItem, newItem) => {
-    this.panelMap[newItem.i!].updateGridPos(newItem);
+    const panel = this.panelMap[newItem.i!];
+    panel.updateGridPos(newItem);
+    panel.configRev++; // trigger change handler
```

Comment:
would ne nicer if incrementing configRev was done by updateGridPos :)

Reply:
I started with that.... but everything gets at least one call to `updateGridPos`, so this was may attempt to only get the resize events :man_shrugging:

Reply:
I think exposing a property/field like `configRev` is a code smell and will lead to all sorts of issues moving forward. Can we encapsulate this some way? I'm aware this PR didn't introduce it 😉

Reply:
Not sure it's related to this PR but `updateQueries` function in https://github.com/grafana/grafana/blob/646d95b8fbd962e8cdefd461fe69f92a05f3ce37/public/app/features/dashboard/state/PanelModel.ts#L438 will always change `configRev` for plugins that upgrade "their" queries on load. We need to solve the upgrade of queries somehow without calling updateQueries to reduce the amount of false save/change popups.

Reply:
> I think exposing a property/field like configRev is a code smell and will lead to all sorts of issues moving forward. Can we encapsulate this some way? I'm aware this PR didn't introduce it 😉

+1

Reply:
> I think exposing a property/field like configRev is a code smell and will lead to all sorts of issues moving forward. Can we encapsulate this some way? I'm aware this PR didn't introduce it 😉

+1